mux.go

```diff
@@ -260,6 +262,14 @@ func (r *Router) SkipClean(value bool) *Router {
 	return r
 }
 
+// OmitRouteFromContext defines the behavior of omitting the Route from the
+//   http.Request context.
+// CurrentRoute will yield nil with this option.
+func (r *Router) OmitRouteFromContext(value bool) *Router {
```

Comment:
Is this function used anywhere outside of tests?

Reply:
Yes, in the users application code. It enables an optional optimization for allocations (3 allocations and 448B per request with no vars, see 1465de4bc6d23ba2349b3823685fb946bfe11d11).

Reply:
Makes sense, thanks! I wish we had an interface for "Router" that would define this contract to make that relationship more clear - however that'll fall outside the scope of this PR 🙂